### PR TITLE
DOC: add note for torch.clamp() special case min > max See #45664

### LINKED ISSUE #45664: What would happen if torch.clamp() has min greater than max?
It seems the current implementation will cut the tensor to min or max according to their values, which might be a problem when debugging logical errors or typos. Maybe it would be better to just throw an exception? Also it would be nice to note that in the [documentation](https://pytorch.org/docs/stable/generated/torch.clamp.html).
```
>>> import torch
>>> print(torch.__version__)
1.6.0
>>> a = torch.randn(4)
>>> a
tensor([-1.1582,  0.0216,  0.6430,  0.5723])
>>> torch.clamp(a, min=0.5, max=-0.5)
tensor([ 0.5000,  0.5000, -0.5000, -0.5000])
```

cc @jlin27 @mruberry

### PR DESCRIPTION
**Pull request by xamm**
Fixes #45664

This PR adds a note to the documentation for `torch.clamp()` to alert users to a special case: If `min` is greater than `max`, all values are set to the `max` value.

Also, an example was added after the first code example. And this one is referenced in the note.

### CODE REVIEW

File: torch/_torch_docs.py [hunk 1]
```diff
@@ -1916,6 +1922,8 @@ def merge_dicts(*dicts):
     tensor([-1.7120,  0.1734, -0.0478, -0.0922])
     >>> torch.clamp(a, min=-0.5, max=0.5)
     tensor([-0.5000,  0.1734, -0.0478, -0.0922])
+    >>> torch.clamp(a, min=8.0, max=11.0)
```
Comment (xamm) [inline torch/_torch_docs.py:1925]: Should there be an example for this special case or is the note sufficient?

File: torch/_torch_docs.py [hunk 2]
```diff
@@ -1901,6 +1901,12 @@ def merge_dicts(*dicts):
     y_i = \min(\max(x_i, \text{min\_value}), \text{max\_value})
 """ + r"""
 
+.. note::
+    If :attr:`min` is greater than :attr:`max` :func:`torch.clamp(..., min, max) <torch.clamp>`
+    sets all values equal to :attr:`max`.
```
Comment (mruberry) [inline torch/_torch_docs.py:1906]: "all values" -> "all output values"?

File: torch/_torch_docs.py [hunk 3]
```diff
@@ -1901,6 +1901,12 @@ def merge_dicts(*dicts):
...
+
+    This is shown in the second example below.
```
Comment (mruberry) [inline torch/_torch_docs.py:1908]: We can probably get rid of this sentence

File: torch/_torch_docs.py (hunk 1 above)
Comment (mruberry) [reply torch/_torch_docs.py:1925]: I think a note alone makes sense because this is a degenerate case we don't want to highlight with an example

**Review by mruberry — COMMENTED** [review]
Thanks for fixing this, @xamm! I made a few small suggestions

File: torch/_torch_docs.py (hunk 2 above)
Comment (xamm) [reply torch/_torch_docs.py:1906]: Maybe "sets all elements in `input` to `max`." Would be better to be more consistent with the general description that `torch.clamp()` 

> Clamp all elements in input into the range...

File: torch/_torch_docs.py (hunk 1 above)
Comment (xamm) [reply torch/_torch_docs.py]: I removed this with the last commit.

File: torch/_torch_docs.py (hunk 3 above)
Comment (xamm) [reply torch/_torch_docs.py]: The reference to the example is deleted with the example itself.

**Review by mruberry — APPROVED** [review]
Neat! Thanks @xamm!

**Comment by xamm** [comment]
I was happy to help.

**Pull request closed without merging**